src/librustc_codegen_llvm/back/write.rs

```diff
@@ -239,6 +239,7 @@ impl<'a> Drop for DiagnosticHandlers<'a> {
     }
 }
 
+#[allow(improper_ctypes)]
```

Comment:
This actually seems like a true positive, this function has no reason to be extern "C" (or even unsafe) AFAICT.

Reply:
I've removed the `extern "C"` from this function and everything appears to still work.